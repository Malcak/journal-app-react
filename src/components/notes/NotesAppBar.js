import React from 'react';

export const NotesAppBar = () => {
  return (
    <div className="notes__appbar">
      <span>28 de Agosto 2020</span>
      <div>
        <button className="btn">picture</button>
        <button className="btn">Save</button>
      </div>
    </div>
  );
};
