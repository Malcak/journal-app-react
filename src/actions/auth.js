import Swal from 'sweetalert2';
import { firebase, googleAuthProvider } from '../firebase/firebaseConfig';
import { types } from '../types/types';
import { notesLogoutClean } from './notes';

export const startLoginEmailPassword = (email, password) => {
  return (dispatch) => {
    dispatch({ type: types.uiStartLoading });
    return firebase
      .auth()
      .signInWithEmailAndPassword(email, password)
      .then(({ user }) => {
        dispatch(login(user.uid, user.displayName));
        dispatch({ type: types.uiFinishLoading });
      })
      .catch((e) => {
        dispatch({ type: types.uiFinishLoading });
        Swal.fire('Error', e.message, 'error');
      });
  };
};

export const startRegisterWithEmailPasswordName = (email, password, name) => {
  return (dispatch) => {
    dispatch({ type: types.uiStartLoading });
    firebase
      .auth()
      .createUserWithEmailAndPassword(email, password)
      .then(async ({ user }) => {
        await user.updateProfile({ displayName: name });
        dispatch(login(user.uid, user.displayName));
        dispatch({ type: types.uiFinishLoading });
      })
      .catch((e) => {
        dispatch({ type: types.uiFinishLoading });
        Swal.fire('Error', e.message, 'error');
      });
  };
};

export const startGoogleLogin = () => {
  return (dispatch) => {
    firebase
      .auth()
      .signInWithPopup(googleAuthProvider)
      .then(({ user }) => {
        dispatch(login(user.uid, user.displayName));
      })
      .catch((e) => {
        Swal.fire('Error', e.message, 'error');
      });
  };
};

export const login = (uid, displayName) => ({
  type: types.login,
  payload: {
    uid,
    displayName,
  },
});

export const startLogout = () => {
  return async (dispatch) => {
    await firebase.auth().signOut();
    dispatch(logout());
    dispatch(notesLogoutClean());
  };
};

export const logout = () => ({
  type: types.logout,
});
