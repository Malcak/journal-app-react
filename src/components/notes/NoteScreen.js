import React, { useEffect, useRef } from 'react';
import { useDispatch, useSelector } from 'react-redux';
import { useForm } from '../../hooks/useForm';
import { NotesAppBar } from './NotesAppBar';
import { activeNote, startDeleting } from '../../actions/notes';

export const NoteScreen = () => {
  const dispatch = useDispatch();
  const { active: note } = useSelector((state) => state.notes);
  const [formValues, handleInputChange, reset] = useForm(note);
  const { title, body } = formValues;

  const activeId = useRef(note.id);

  useEffect(() => {
    if (note.id !== activeId.current) {
      reset(note);
      activeId.current = note.id;
    }
  }, [note, reset]);

  useEffect(() => {
    dispatch(activeNote(formValues.id, { ...formValues }));
  }, [formValues, dispatch]);

  const handleDelete = () => {
    dispatch(startDeleting(note.id));
  };

  return (
    <div className="notes__main-content">
      <NotesAppBar />
      <div className="notes__content">
        <input
          type="text"
          name="title"
          placeholder="Some awesome title!"
          className="notes__title-input"
          value={title}
          onChange={handleInputChange}
          autoComplete="off"
        />
        <textarea
          name="body"
          placeholder="What happened today..."
          className="notes__textarea"
          value={body}
          onChange={handleInputChange}
        ></textarea>
        {note.imgUrl && (
          <div className="notes__image">
            <img src={note.imgUrl} alt="noteImage" />
          </div>
        )}
      </div>
      <button className="btn btn-danger" onClick={handleDelete}>
        Remove
      </button>
    </div>
  );
};
