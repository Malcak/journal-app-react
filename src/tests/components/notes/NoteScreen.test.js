import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { mount } from 'enzyme';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';
import { NoteScreen } from '../../../components/notes/NoteScreen';
import { activeNote } from '../../../actions/notes';
import notes from '../../fixtures/notes';

jest.mock('../../../actions/notes', () => ({
  activeNote: jest.fn(),
}));

const middlewares = [thunk];
const mockStore = configureStore(middlewares);

const defaultState = {
  auth: {},
  ui: {
    loading: false,
    msgError: null,
  },
  notes: {
    active: notes[0],
    notes: notes,
  },
};

let store = mockStore(defaultState);
store.dispatch = jest.fn();

const wrapper = mount(
  <Provider store={store}>
    <MemoryRouter>
      <NoteScreen />
    </MemoryRouter>
  </Provider>
);

describe('test on <NoteScreen /> component', () => {
  test('should display correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should trigger the change of the active note', () => {
    wrapper
      .find('input[name="title"]')
      .simulate('change', { target: { name: 'title', value: 'Hello Again' } });

    expect(activeNote).toHaveBeenCalledWith(notes[0].id, {
      ...notes[0],
      title: 'Hello Again',
    });
  });
});
