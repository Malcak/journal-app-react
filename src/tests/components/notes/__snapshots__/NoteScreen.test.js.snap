// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test on <NoteScreen /> component should display correctly 1`] = `
<div
  className="notes__main-content"
>
  <div
    className="notes__appbar"
  >
    <span>
      28 de Agosto 2020
    </span>
    <input
      id="fileSelector"
      name="file"
      onChange={[Function]}
      style={
        Object {
          "display": "none",
        }
      }
      type="file"
    />
    <div>
      <button
        className="btn"
        onClick={[Function]}
      >
        Picture
      </button>
      <button
        className="btn"
        onClick={[Function]}
      >
        Save
      </button>
    </div>
  </div>
  <div
    className="notes__content"
  >
    <input
      autoComplete="off"
      className="notes__title-input"
      name="title"
      onChange={[Function]}
      placeholder="Some awesome title!"
      type="text"
      value="Hello World"
    />
    <textarea
      className="notes__textarea"
      name="body"
      onChange={[Function]}
      placeholder="What happened today..."
      value="And welcome to a new video..."
    />
    <div
      className="notes__image"
    >
      <img
        alt="noteImage"
        src="https://res.cloudinary.com/malcak-media-cloud/image/upload/v1626378813/htjjxjjndcweyzl1zesr.jpg"
      />
    </div>
  </div>
  <button
    className="btn btn-danger"
    onClick={[Function]}
  >
    Remove
  </button>
</div>
`;
